### PR TITLE
[SPARK-29653][SQL] Fix MICROS_PER_MONTH in IntervalUtils

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

MICROS_PER_MONTH = DAYS_PER_MONTH * MICROS_PER_DAY

### Why are the changes needed?

fix bug

### Does this PR introduce any user-facing change?

no
### How was this patch tested?

add ut

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan @dongjoon-hyun

File: sql/core/src/test/resources/sql-tests/results/date_part.sql.out [hunk 1]
```diff
@@ -857,7 +857,7 @@ select date_part('epoch', c) from t2
 -- !query 106 schema
 struct<date_part('epoch', t2.`c`):decimal(18,6)>
 -- !query 106 output
-31873892788.332003
+31897220765.004003
```
Comment (dongjoon-hyun) [inline sql/core/src/test/resources/sql-tests/results/date_part.sql.out]: Thanks, @yaooqinn . This looks enough. Shall we revert the changes on `IntervalUtilsSuite.scala`?

Comment (yaooqinn) [reply sql/core/src/test/resources/sql-tests/results/date_part.sql.out]: OK

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/test/resources/sql-tests/results/date_part.sql.out (hunk 1 above)
Comment (MaxGekk) [inline sql/core/src/test/resources/sql-tests/results/date_part.sql.out]: I have compared to PostgreSQL output:
```sql
maxim=# select date_part('epoch', interval '1010 year 9 month 8 day 7 hour 6 minute 5 second 4 millisecond 3 microseconds');
    date_part
-----------------
 31897220765.004
(1 row)
```
It seems PostgreSQL just looses the precision.

**Review by MaxGekk — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you all.
Merged to master.
